staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -334,7 +334,12 @@ func (s *APIAggregator) PrepareRun() (preparedAPIAggregator, error) {
 	// add post start hook before generic PrepareRun in order to be before /healthz installation
 	if s.openAPIConfig != nil {
 		s.GenericAPIServer.AddPostStartHookOrDie("apiservice-openapi-controller", func(context genericapiserver.PostStartHookContext) error {
-			go s.openAPIAggregationController.Run(context.StopCh)
+			initialOpenAPIAggrDoneCh := make(chan struct{})
+			go s.openAPIAggregationController.Run(context.StopCh, initialOpenAPIAggrDoneCh)
+			select {
+			case <-context.StopCh:
+			case <-initialOpenAPIAggrDoneCh:
```

Comment:
It is ensured that the APIService informer is ready here? Otherwise, we won't see all api service names.

I believe the `AggregationController.AddAPIService` func is called via informer events and it registers the api services asynchronously.